File: api/src/main/java/com/bindglam/goldengine/manager/Context.java
package com.bindglam.goldengine.manager;

import com.bindglam.goldengine.GoldEngineConfiguration;
import com.bindglam.goldengine.GoldEnginePlugin;
import org.jetbrains.annotations.NotNull;

import java.util.logging.Logger;

/**
 * Context interface for managers
 *
 * @author bindglam
 */
public interface Context {
    @NotNull GoldEnginePlugin plugin();

    @NotNull GoldEngineConfiguration config();

    @NotNull Logger logger();
}

File: core/src/main/kotlin/com/bindglam/mint/manager/Context.kt
package com.bindglam.mint.manager

import com.bindglam.mint.MintConfiguration
import com.bindglam.mint.MintPlugin
import com.bindglam.mint.utils.plugin

class Context(private val plugin: MintPlugin) {
    fun plugin() = this.plugin
    fun config(): MintConfiguration = this.plugin.plugin().mintConfig
    fun logger() = this.plugin.plugin().logger
}